api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java

```diff
@@ -241,12 +249,21 @@ private WorkspaceAdminView getActiveWorkspaceAdminView(
             .getWorkspaceAsService(workspaceNamespace, workspaceFirecloudName)
             .getWorkspace();
 
-    return new WorkspaceAdminView()
-        .workspace(workspaceMapper.toApiWorkspace(dbWorkspace, firecloudWorkspace))
-        .workspaceDatabaseId(dbWorkspace.getWorkspaceId())
-        .collaborators(collaborators)
-        .resources(adminWorkspaceResources)
-        .activeStatus(dbWorkspace.getWorkspaceActiveStatusEnum());
+    return workspaceConfigProvider.get().featureFlags.enablePublishedWorkspacesViaDb
+        ? new WorkspaceAdminView()
+            .workspace(
+                workspaceMapper.toApiWorkspace(
+                    dbWorkspace, firecloudWorkspace, featuredWorkspaceService))
+            .workspaceDatabaseId(dbWorkspace.getWorkspaceId())
+            .collaborators(collaborators)
+            .resources(adminWorkspaceResources)
+            .activeStatus(dbWorkspace.getWorkspaceActiveStatusEnum())
+        : new WorkspaceAdminView()
+            .workspace(workspaceMapper.toApiWorkspace(dbWorkspace, firecloudWorkspace))
+            .workspaceDatabaseId(dbWorkspace.getWorkspaceId())
+            .collaborators(collaborators)
+            .resources(adminWorkspaceResources)
+            .activeStatus(dbWorkspace.getWorkspaceActiveStatusEnum());
```

Comment:
It looks like these conditions only vary on .workspace.

Do you think its worth pulling the new WorkspaceAdminView().workspace(...) into a conditional variable declaration and then assign the common attributes to the variable. Then return the variable?

Reply:
I could, i felt this would be easier to delete once we will do clean up of flag. I'll update this